exe/task.cc

```diff
@@ -137,6 +148,161 @@ static map<string, string> generate_environment_from(lth_jc::JsonContainer &inpu
     return env_vars;
 }
 
+// TODO: PCP-769 added routines start here!
+
+// Take out common permissions.
+fs::perms TASK_FILE_RW_PERMS = fs::owner_read | fs::owner_write | fs::group_read | fs::group_write;
+
+// Creates the SYSTEM_PREFIX/pxp-agent/tasks-cache/<sha> directory,
+// and ensures that its permissions are only readable/writable by
+// the PXP agent owner/group. Returns the path to this directory.
+// Note that the last modified time of the directory is updated.
+/*static*/ fs::path create_tasks_cache_dir(const string &sha256) {
+    auto tasks_cache_dir = fs::path(SYSTEM_PREFIX()) / "pxp-agent" / "tasks-cache" / sha256;
+    try {
+      fs::create_directories(tasks_cache_dir);
+      fs::last_write_time(tasks_cache_dir, time(nullptr)); 
+      fs::permissions(tasks_cache_dir, TASK_FILE_RW_PERMS);
```

Comment:
Inheritance does sound right. The right thing to do would then be to have packaging create the pxp-agent/spool and pxp-agent/tasks directories, and enforce more restricted permissions on them?